src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferAsSpanOverSubstring.cs

```diff
@@ -0,0 +1,298 @@
+            foreach (var candidate in candidates)
+            {
+                int quality = EvaluateCandidateQuality(symbols, isSubstringLookup, invocation, candidate);
+
+                //  Reject candidates that do not replace at least one Substring call.
+                if (quality < 1)
+                {
+                    continue;
+                }
+                else if (quality == resultQuality)
+                {
+                    bestCandidates.Add(candidate);
+                }
+                else if (quality > resultQuality)
+                {
+                    resultQuality = quality;
+                    bestCandidates.Clear();
+                    bestCandidates.Add(candidate);
+                }
+            }
+
+            return bestCandidates.ToImmutable();
+
+            //  Returns a number indicating how good the candidate method is. 
+            //  If the candidate is valid, the number of Substring calls that can be replaced with AsSpan calls is returned.
+            //  If the candidate is invalid, -1 is returned.
+            static int EvaluateCandidateQuality(in RequiredSymbols symbols, ReadOnlySpan<bool> isSubstringLookup, IInvocationOperation invocation, IMethodSymbol candidate)
+            {
+                var method = invocation.TargetMethod;
+
+                if (candidate.Parameters.Length != method.Parameters.Length)
+                    return -1;
+
+                int replacementCount = 0;
+                foreach (var parameter in candidate.Parameters)
+                {
+                    if (isSubstringLookup[parameter.Ordinal] && SymbolEqualityComparer.Default.Equals(parameter.Type, symbols.ReadOnlySpanOfCharType))
+                    {
+                        ++replacementCount;
+                        continue;
+                    }
+
+                    var oldParameter = method.Parameters[parameter.Ordinal];
+                    if (!SymbolEqualityComparer.Default.Equals(parameter.Type, oldParameter.Type))
+                    {
+                        return -1;
+                    }
+                }
+
+                return replacementCount;
+            }
+        }
+
+        private static IEnumerable<IMethodSymbol> GetAllAccessibleOverloadsAtInvocationCallSite(IInvocationOperation invocation, CancellationToken cancellationToken)
+        {
+            var method = invocation.TargetMethod;
+            var model = invocation.SemanticModel;
+            int location = invocation.Syntax.SpanStart;
+            var instance = invocation.Instance;
+
+            IEnumerable<IMethodSymbol> allOverloads;
+            if (method.IsStatic)
+            {
+                allOverloads = model.LookupStaticMembers(location, method.ContainingType, method.Name).OfType<IMethodSymbol>();
+            }
+            //  Ensure protected members can only be invoked on instances that are known to be instances of the accessing class.
```

Comment:
Please do no add comments between and if and else blocks. Can you either move the comment at start of the if-else OR inside the else block?

Reply:
Fixed.